Feature/smhe 1105 periodic pq read voor dsmr4 smr5 devices public and private

natasja-n opened the pull request.

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
w.i.p.

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
Merge branch 'feature/SMHE-1103_actual_pq' into feature/SMHE-1105_Periodic_PQ_Read_voor_DSMR4_SMR5_devices_Public_and_Private

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
w.i.p.

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
w.i.p.

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
w.i.p.

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
w.i.p.

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
Merge branch 'feature/SMHE-1103_actual_pq' into feature/SMHE-1105_Periodic_PQ_Read_voor_DSMR4_SMR5_devices_Public_and_Private

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
update AbstractGetPowerQualityProfileHandler with ObjectConfigService

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
update AbstractGetPowerQualityProfileHandler with ObjectConfigService

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

stefanermens committed:
SMHE-1105: Refactor how scaler_unit is used

Signed-off-by: stefanermens <stefan.ermens@alliander.com>

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
add testHandlePublicOrPrivateProfileWithoutSelectiveAccess, ObjectConfigServiceHelper

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
add testHandlePublicOrPrivateProfileWithoutSelectiveAccess, ObjectConfigServiceHelper

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
add testHandlePublicOrPrivateProfileWithoutSelectiveAccess, ObjectConfigServiceHelper

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
add testHandlePublicOrPrivateProfileWithoutSelectiveAccess, ObjectConfigServiceHelper

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
w.i.p. GetPowerQualityProfileSelectiveAccessHandlerTest

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

stefanermens committed:
SMHE-1105: Get selectable objects from profile

Signed-off-by: stefanermens <stefan.ermens@alliander.com>

stefanermens committed:
SMHE-1105: Add explanation and fix getting the selectable objects in a profile

Signed-off-by: stefanermens <stefan.ermens@alliander.com>

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
add GetPowerQualityProfileSelectiveAccessHandlerTest

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
Merge branch 'development' into feature/SMHE-1105_Periodic_PQ_Read_voor_DSMR4_SMR5_devices_Public_and_Private

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
add GetPowerQualityProfileSelectiveAccessHandlerTest

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
remove TODO, make getIntervalInMinutes dynamic

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-dlms/src/main/java/org/opensmartgridplatform/dlms/services/ObjectConfigService.java:
```diff
@@ -67,6 +67,31 @@ public CosemObject getCosemObject(
+          } else {
+            throw new IllegalArgumentException(
+                String.format(
+                    "No object found of type %s in profile %s version %s",
```
is het profile of protocol?

natasja-n replied:
Hij is geupdate naar protocol.

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/resources/dlmsprofiles/dlmsprofile-smr50.json:
```diff
@@ -211,7 +211,7 @@
       "class-id": 7,
       "version": 1,
-      "obis": "0.1.99.1.2.255",
+      "obis": "1.0.99.1.2.255",
```
Bewust omgedraaid?

natasja-n replied:
ja, hij stond verkeerd.

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
remove TODO, make getIntervalInMinutes dynamic

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/testutil/ObjectConfigServiceHelper.java:
```diff
@@ -0,0 +1,87 @@
+    if (polyphase) {
+      return Collections.singletonList(MeterType.PP);
+    } else {
+      return Arrays.asList(MeterType.SP, MeterType.SP);
```
Staat hier bewust 2 keer zelfde enum?

natasja-n replied:
nee, niet bewust, nu worden beide typen in de lijst gezet.

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/testutil/ObjectConfigServiceHelper.java:
```diff
@@ -0,0 +1,87 @@
+
+    final CosemObject clock = new CosemObject();
+    clock.setClassId(8);
+    clock.setObis("0.0.1.0.0.255");
```
Willen we de OBIS codes niet in constants hebben?

natasja-n replied:
ja, beter, staat nu in een contstante.

stefanermens committed:
SMHE-1105: Fix obis codes for M-Bus objects

Signed-off-by: stefanermens <stefan.ermens@alliander.com>

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
update interval DEFINABLE_LOAD_PROFILE to 900sec

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
Merge branch 'feature/SMHE-1105_Periodic_PQ_Read_voor_DSMR4_SMR5_devices_Public_and_Private' of github.com:OSGP/open-smart-grid-platform into feature/SMHE-1105_Periodic_PQ_Read_voor_DSMR4_SMR5_devices_Public_and_Private

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
process review remarks: protocol in logging error, update metertypes enum in ObjectConfigServiceHelper, use constant for Obis code

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1105 Periodic PQ Read voor DSMR4 SMR5 devices - Public and Private
update interval in config for dsmr4 as well

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>
